src/rust/lib/graph-editor/src/component/visualization/traits.rs

```diff
@@ -0,0 +1,92 @@
+//! Provides traits that let us know about the capabilities of complex UI components..
+
+use crate::prelude::*;
+
+use ensogl::display::Scene;
+use ensogl::display::Symbol;
+
+
+
+// ==================================
+// === UI Component Helper Traits ===
+// ==================================
+
+/// Indicates the desired target layer.
+#[derive(Copy,Clone,Debug)]
+pub enum TargetLayer {
+    /// A symbol that goes onto the main layer.
+    Main,
+    /// A visualisation symbol that goes into the visualisation layer.
+    Visualisation,
+}
+
+/// Contains a symbol and information about which layer it should be placed on.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+pub struct SymbolLayoutData {
```

Comment:
I still have problem with the name. It doesn't tell me what it is. I would vote for `SymbolWithLayout` instead.